eureka-core/src/main/java/com/netflix/eureka/PeerAwareInstanceRegistry.java

```diff
@@ -209,9 +208,14 @@ public void run() {
      */
     private void updatePeerEurekaNodes() {
         InstanceInfo myInfo = ApplicationInfoManager.getInstance().getInfo();
-        List<String> replicaUrls = DiscoveryManager.getInstance()
-                .getDiscoveryClient()
-                .getDiscoveryServiceUrls(DiscoveryClient.getZone(myInfo));
+
+        EurekaClientConfig clientConfig = DiscoveryManager.getInstance().getEurekaClientConfig();
```

Comment:
Agreed. But let's do this in a separate pull and not conflate the current changes.